src/compiler/parser.ts

```diff
@@ -8801,6 +8801,7 @@ namespace Parser {
 
             const comment = doInsideOfContext(NodeFlags.JSDoc, () => parseJSDocCommentWorker(start, length));
             setParent(comment, parent);
+            setParentRecursive(comment, /*incremental*/ false);

```

Comment:
I know it's an added cost but I'm also hesitant to add `setParent` all over the place. Maybe this should be moved somewhere within services? but then it would require a full crawl of the program to find JSDoc comments so nodes within them could get their `.parent` set.

Regular nodes receive their `.parent` from the binder when the typechecker gets created but that's not the case for JSDoc